deeppavlov/models/embedders/tfidf_weighted_embedder.py

```diff
@@ -75,18 +87,12 @@ def __init__(self,
                  idf_base_count: int = 100,
                  log_base: int = 10,
                  min_idf_weight=0.0, **kwargs) -> None:
-        """
-        Initialize embedder with given parameters.
-        """
         self.embedder = embedder
         self.dim = self.embedder.dim
         self.mean = mean
         self.pad_zero = pad_zero
-
-        if tokenizer is None:
-            self.tokenizer = self.space_detokenizer
-        else:
-            self.tokenizer = tokenizer
+        self.tokenizer = tokenizer or self.space_detokenizer
+        self.vectorizer = None
```

Comment:
```suggestion
        self.vectorizer = vectorizer
```
if `vectorizer` is `None`, then it's `None`
And then remove `self.vectorizer = vectorizer` in `elif`